### PR TITLE
Fix/routing

### PR DESCRIPTION
**Pull request by IsraaSulaiman**
## Main Changes
- Make another high order component: 
  - when a user (who is logged in) make requests to login, register or `/`, he will redirected to `/home`
- Add a function `setUserName` to store the username in the app state and pass it to the routes
- Add condition to the `navLinksForUsers.map` so that they will only be rendered when the user is logged to handle his requests.

## Note: 
This PR includes the commits of my previous PR. They will be merged forcebelly once the `fix/style` PR is merged.

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,69 +1,86 @@
...
-              exact
-              path={"/"}
-              // navLinksForUsers is an array with home as the first item
-              component={navLinksForUsers[0].component}
-            />
-          ) : null}
+          <Route
+            exact
+            path="/about-us"
+            component={AboutUs}
```
Comment (dubhcait) [inline client/src/App.js]: the about us needs the   render={props => (
                <AboutUs isLogged={IsLogged} {...props} />
              )

for the header component actually so does the 404 page on line 80

File: client/src/Component/Pages/Status/index.js [hunk 2]
```diff
@@ -119,7 +117,7 @@ export default class Status extends Component {
         <Header isLogged={true} />
         <div className="status">
           <h3 className="status__greeting">
-            Hi <span className="status__username">{name}</span>
+            Hi <span className="status__username">{this.props.username}</span>
```
Comment (dubhcait) [inline client/src/Component/Pages/Status/index.js]: I was just thinking about this, well done

File: client/src/Component/Pages/Signup/index.js [hunk 3]
```diff
@@ -134,17 +132,19 @@ export default class SignUp extends Component {
...
-          })
-          .catch(error => {
-            notification(
-              this.notificationDOMRef,
-              'warning',
-              'Sorry, something went wrong. Please try again!',
+              );
+              // Success
+              this.props.setIsLogged(true);
+              this.props.setUserName(result.data.username)
```
Comment (dubhcait) [inline client/src/Component/Pages/Signup/index.js]: Could we set this to their nickName' as we asked them how they would like to be called?

File: client/src/App.js (hunk 1 above)
Comment (IsraaSulaiman) [reply client/src/App.js]: Aha. Will fix that

File: client/src/Component/Pages/Signup/index.js (hunk 3 above)
Comment (IsraaSulaiman) [reply client/src/Component/Pages/Signup/index.js]: Ya sure

File: src/controller/checkLogin.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
 module.exports = (req, res) => {
   //User is authenticated
-  res.send({ auth: 'ok' });
+  res.send({ auth: 'ok', username: req.auth.username });
```
Comment (dubhcait) [inline src/controller/checkLogin.js]: love your solution to this!

**Pull request merged**